### PR TITLE
Loading Spinners

### PR DESCRIPTION
**Pull request by jwu910**
Add loading spinners to Missions pages at the list level so the page header is still visible. Added in a style injection for MUI circular progress component at the app level to center circular progress components. Please let me know if you have any questions. #121

### CODE REVIEW

File: src/app/page/Missions/Missions.jsx [hunk 1]
```diff
@@ -40,41 +40,41 @@ const MissionsPage = ({ user, history, firebase, ...rest }) => {
...
+                  history.push(`/missions/${mission.id}`);
+                }}
+              >
+                Details
+              </StyledButton>
+            </Grid>
+          </Card>
+        ))
+      ) : (
+        <CircularProgress />
```
Comment (thedma04) [inline src/app/page/Missions/Missions.jsx:76]: This is good, but in a situation where there are errors fetching missions, the loading will be there forever until refreshed. Do we wanna keep a loading and error state to mitigate this behaviour??

File: src/app/page/MissionsCreated/Missions.jsx [hunk 2]
```diff
@@ -25,24 +25,28 @@ const MissionsPage = ({ auth, history, firebase, ...rest }) => {
...
+                  history.push(`/missions/${mission.id}`);
+                }}
+              >
+                Details
+              </StyledButton>
+            </Grid>
+          </Card>
+        ))
+      ) : (
+        <CircularProgress />
```
Comment (thedma04) [inline src/app/page/MissionsCreated/Missions.jsx:48]: The same applies to this

Comment (thedma04) [inline src/app/page/MissionsCreated/Missions.jsx:48]: The same applies to this

File: src/app/page/Missions/Missions.jsx (hunk 1 above)
Comment (azmainamin) [reply src/app/page/Missions/Missions.jsx:76]: I think we have to.  Maybe create a separate issue for that?

Comment (jwu910) [reply src/app/page/Missions/Missions.jsx:76]: Correct, Felt that any error handling would creep the scope of this ticket. Was planning to bring that up in another issue.

The missions object looks to be assigned through a useSelector hook(?) that I'm I hadn't had a chance to get familiar with yet.

Would you advise to change the scope of the ticket to include error handling for the api call?

Comment (thedma04) [reply src/app/page/Missions/Missions.jsx:76]: Can you create an issue for that whilst I approve this

**Pull request merged**